docs/faq/ImmutableData.md

```diff
@@ -0,0 +1,245 @@
+<a id="issues-with-es6-for-immutable-ops"></a>
+## What are the issues with using ES6 for immutable operations?
+ES6 was never designed to provide guaranteed immutable operations. Accordingly, there are several issues you need to be aware of if you choose to use ES6 for your Immutable operations in your Redux app.
+#### Accidental Object Mutation
+With ES6, you can accidentally mutate an object (such as the Redux state tree) quite easily without realising it. For example, updating deeply nested properties, creating a new *reference* to an object instead of a new object, or performing a shallow copy rather than a deep copy, can all lead to inadvertent object mutations, and can trip up even the most experienced JavaScript coder. 
+
+To avoid these issues, ensure you follow the recommended [Immutable Update Patterns for ES6](http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html).
+
+#### Verbose Code
+Updating complex nested state trees can lead to verbose code that is tedious to write and difficult to debug.
+
+#### Poor Performance
+Operating on ES6 objects and arrays in an immutable way can be slow, particularly as your state tree grows larger.
+
+**Documentation**
+- [Immutable Update Patterns for ES6](http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html)
+
+<a id="why-use-immutable-library"></a>
+## Why should I use an immutable-focused library such as Immutable.JS?
+
+Immutable-focused libraries such as Immutable.js have been designed to overcome the issues with immutability inherent within ES6, providing all the benefits of immutability with the performance your app requires. Whether you choose to use such a library, or stick with ES6, depends on how comfortable you are with adding another dependency to your app, or how sure you are that you can avoid the pitfalls inherent within ES6’s approach to immutability.
+
+Whichever option you choose, make sure you’re familiar with the concepts of [immutability, side effects and mutation](http://redux.js.org/docs/recipes/reducers/PrerequisiteConcepts.html#note-on-immutability-side-effects-and-mutation). In particular, ensure you have a deep understanding of what ES6 does when updating and copying values in order to guard against accidental mutations that will degrade you app’s performance, or break it altogether.
+
+**Documentation**
+- [immutability, side effects and mutation](http://redux.js.org/docs/recipes/reducers/PrerequisiteConcepts.html#note-on-immutability-side-effects-and-mutation)
```

Comment:
Sorry about that. I'll go through the whole PR to ensure the formatting follows the existing convention once you're happy with the changes.